src/app/modal-dialog/nested-modal-dialog.component.ts

```diff
@@ -44,12 +40,12 @@ export class NestedModalDialogComponent {
 
   openModel() {
     this.dialog
-      .modal(NestedModalDialogComponent, this.placeholder)
+      .modal(NestedModalDialogComponent)
       .title(`Modal Dialog no. '${++NestedModalDialogComponent.s_depth}'.`)
       .buttons(
         [{text: 'Cancel', click: (e, modal) => { modal.close(true); }},
          {text: 'OK', click: (e, modal) => { modal.close(true); }, isDefault: true}])
-      .afterClose((f) => { NestedModalDialogComponent.s_depth--; })
-      .open();
+      .open()
+      .afterClose((f) => { NestedModalDialogComponent.s_depth--; });
```

Comment:
afterClose cannot be called before open

Reply:
This would be subtle breaking change - has something changed in how this is implemented, previously I had an `Subject` to handle that.